lupa/_lupa.pyx

```diff
@@ -1462,6 +1463,19 @@ cdef int py_to_lua(LuaRuntime runtime, lua_State *L, object o, bint wrap_none=Fa
     elif isinstance(o, float):
         lua.lua_pushnumber(L, <lua.lua_Number><double>o)
         pushed_values_count = 1
+    elif recursive and isinstance(o, Sequence):
+        lua.lua_createtable(L, <int>len(o), 0)   # create a table at the top of stack, with narr already known
+        for i, v in enumerate(o):
+            py_to_lua(runtime, L, v, wrap_none, recursive)
+            lua.lua_rawseti(L, -2, i+1)
+        pushed_values_count = 1
+    elif recursive and isinstance(o, Mapping):
+        lua.lua_createtable(L, 0, <int>len(o))  # create a table at the top of stack, with nrec already known
+        for key, value in o.iteritems(): # to compatible with py2
+            py_to_lua(runtime, L, key, wrap_none, recursive)
+            py_to_lua(runtime, L, value, wrap_none, recursive)
+            lua.lua_rawset(L, -3)
+        pushed_values_count = 1
     else:
         if isinstance(o, _PyProtocolWrapper):
```

Comment:
We should keep this case early. It's a cheap type check compared to the ABCs `Sequence` and `Mapping`, and not so unlikely to happen.